add-on/panels/js/panel.jsx

```diff
@@ -894,6 +885,114 @@ loop.panel = (function(_, mozL10n) {
     }
   });
 
+  var SharePanelView = React.createClass({
+    mixins: [
+      Backbone.Events,
+      sharedMixins.DocumentVisibilityMixin,
+      sharedMixins.WindowCloseMixin
+    ],
+
+    propTypes: {
+      dispatcher: React.PropTypes.instanceOf(loop.Dispatcher).isRequired,
+      store: React.PropTypes.instanceOf(loop.store.RoomStore).isRequired
+    },
+
+    getInitialState: function() {
+      return this.props.store.getStoreState();
+    },
+
+    componentDidMount: function() {
+      this.listenTo(this.props.store, "change", this._onStoreStateChanged);
+    },
+
+    componentWillUnmount: function() {
+      this.stopListening(this.props.store);
+    },
+
+    _onStoreStateChanged: function() {
+      this.setState(this.props.store.getStoreState());
+    },
+
+    onDocumentHidden: function() {
+      this.state.shouldOpenSharePanel && this._closeSharePanel();
+    },
+
+    componentWillUpdate: function(nextProps, nextState) {
+      // If we've just created a room, close the panel - the store will open
+      // the room.
+      if (this.state.pendingCreation &&
+          !nextState.pendingCreation && !nextState.error) {
+        this.setState({
+          shouldOpenSharePanel: true
+        });
+      }
+    },
+
+    _closeSharePanel: function() {
+      this.setState({
+        shouldOpenSharePanel: false
+      });
+
+      this.openRoom();
+    },
+
+    openRoom: function() {
+      var activeRoom = this.state.activeRoom;
+      this.props.dispatcher.dispatch(new sharedActions.OpenRoom({
+        roomToken: activeRoom.roomToken
+      }));
+
+      this.closeWindow();
+    },
+
+    render: function() {
+      var sharePanelClasses = classNames({
+        "share-panel-container": true,
+        "share-panel-open": this.state.shouldOpenSharePanel
+      });
+      var activeRoom = this.state.activeRoom;
+      var roomData = {
+        roomUrl: activeRoom.roomUrl,
+        roomContextUrls: (activeRoom.decryptedContext && activeRoom.decryptedContext.urls) || []
+      };
+
+      return (
+        <div className={sharePanelClasses}>
+          <div className="share-panel-overlay" onClick={this._closeSharePanel} />
+          <div className="share-panel-content">
+            <h2>{mozL10n.get("share_panel_heading")}</h2>
+            <p>
+              {mozL10n.get("share_panel_subheading")}
+            </p>
+
+            <div className="share-panel-link">
+              <span>{mozL10n.get("share_panel_link")}</span>
+              <input type="text" value={activeRoom.roomUrl} />
+              <sharedViews.CopyLinkButton
```

Comment:
I think its good that you've shared the buttons, but I think we should go a bit further. I think we should share probably all of `share-panel-content`.

The layout should be identical - it might be worth checking with UX if they really do want different text sizes etc between the two or not. If we do need to have slightly different sized text / icons then we should do that as properties (e.g. textSize="small") on the react element, translating to css classes that are then styled appropriately for the component.

I know the link text box is different, but I don't see a reason for it to be different between them - we should keep the same structure.